test/refactoring/ConvertAsync/convert_to_continuation.swift

```diff
@@ -0,0 +1,351 @@
+  _ = withoutAsyncAlternativeBecauseOfReturnValue {
+    completionHandler($0)
+  }
+}
+// CREATE-CONTINUATION-BECAUSE-RETURN-VALUE:      func testCreateContinuationBecauseOfReturnValue() async -> Int {
+// CREATE-CONTINUATION-BECAUSE-RETURN-VALUE-NEXT:   return await withCheckedContinuation { (continuation: CheckedContinuation<Int, Never>) in 
+// CREATE-CONTINUATION-BECAUSE-RETURN-VALUE-NEXT:     _ = withoutAsyncAlternativeBecauseOfReturnValue {
+// CREATE-CONTINUATION-BECAUSE-RETURN-VALUE-NEXT:       continuation.resume(returning: $0)
+// CREATE-CONTINUATION-BECAUSE-RETURN-VALUE-NEXT:     }
+// CREATE-CONTINUATION-BECAUSE-RETURN-VALUE-NEXT:   }
+// CREATE-CONTINUATION-BECAUSE-RETURN-VALUE-NEXT: }
+
+
+// RUN: %refactor-check-compiles -convert-to-async -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=THROWING-CONTINUATION %s
+func testThrowingContinuation(completionHandler: (Int?, Error?) -> Void) {
+  withoutAsyncAlternativeThrowing { (theValue, theError) in
+    if let theError = theError {
+      completionHandler(nil, theError)
+    } else {
+      completionHandler(theValue!, nil)
+    }
+  }
+}
+// THROWING-CONTINUATION:      func testThrowingContinuation() async throws -> Int {
+// THROWING-CONTINUATION-NEXT:   return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Int, Error>) in 
+// THROWING-CONTINUATION-NEXT:     withoutAsyncAlternativeThrowing { (theValue, theError) in
+// THROWING-CONTINUATION-NEXT:     if let theError = theError {
+// THROWING-CONTINUATION-NEXT:       continuation.resume(throwing: theError)
+// THROWING-CONTINUATION-NEXT:     } else {
+// THROWING-CONTINUATION-NEXT:       continuation.resume(returning: theValue!)
+// THROWING-CONTINUATION-NEXT:     }
+// THROWING-CONTINUATION-NEXT:   }
+// THROWING-CONTINUATION-NEXT: }
+
+// We can't relay both the result and the error through the continuationinuation. Converting the following results in a compiler error complaining that theError (of type Error?) can't be passed to `continuation.resume(throwing)`.
+// RUN: %refactor -convert-to-async -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=THROWING-CONTINUATION-RELAYING-ERROR-AND-RESULT %s
+func testThrowingContinuationRelayingErrorAndResult(completionHandler: (Int?, Error?) -> Void) {
+  withoutAsyncAlternativeThrowing { (theValue, theError) in
+    completionHandler(theValue, theError)
+  }
+}
+// THROWING-CONTINUATION-RELAYING-ERROR-AND-RESULT:      func testThrowingContinuationRelayingErrorAndResult() async throws -> Int {
+// THROWING-CONTINUATION-RELAYING-ERROR-AND-RESULT-NEXT:   return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Int, Error>) in 
+// THROWING-CONTINUATION-RELAYING-ERROR-AND-RESULT-NEXT:     withoutAsyncAlternativeThrowing { (theValue, theError) in
+// THROWING-CONTINUATION-RELAYING-ERROR-AND-RESULT-NEXT:       continuation.resume(throwing: theError)
```

Comment:
Would it be possible to check for this case specifically and add our own handling? Ie. convert this into:
```
if let err = theError {
  continuation.resume(throwing: theError)
} else {
  continuation.resume(returning: theValue) // or (a, b, c, d) if multiple
}
```

Reply:
After discussing this with you, I'm happy to just leave the ambiguous case to another PR. We have a similar problem with the regular hoisting where we end up with `throw <#err#>`. This is better than what we have at the moment anyway.